tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,415 @@
+        (client, container, prefix) = self.policy_blob_client
+
+        try:
+            # All blobs with YAML extension
+            blobs = [b for b in client.list_blobs(container) if Host.has_yaml_ext(b.name)]
+        except AzureHttpError as e:
+            # If blob methods are failing don't keep
+            # a cached client
+            self.policy_blob_client = None
+            raise e
+
+        # Filter to hashes we have not seen before
+        new_blobs = [b for b in blobs
+                     if b.properties.content_settings.content_md5 != self.blob_cache.get(b.name)]
+
+        # Get all YAML files on disk that are no longer in blob storage
+        cached_policy_files = [f for f in os.listdir(self.policy_cache)
+                               if Host.has_yaml_ext(f)]
+
+        removed_files = [f for f in cached_policy_files if f not in [b.name for b in blobs]]
+
+        if not (removed_files or new_blobs):
+            return
+
+        # Update a copy so we don't interfere with
+        # iterations on other threads
+        policies_copy = self.policies.copy()
+
+        for f in removed_files:
+            path = os.path.join(self.policy_cache, f)
+            self.unload_policy_file(path, policies_copy)
+
+        # Get updated YML files
+        for blob in new_blobs:
+            policy_path = os.path.join(self.policy_cache, blob.name)
+            if os.path.exists(policy_path):
+                self.unload_policy_file(policy_path, policies_copy)
+
+            client.get_blob_to_path(container, blob.name, policy_path)
+            self.load_policy(policy_path, policies_copy)
+            self.blob_cache.update({blob.name: blob.properties.content_settings.content_md5})
+
+        # Assign our copy back over the original
+        self.policies = policies_copy
+
+        if self.require_event_update:
+            self.update_event_subscriptions()
+
+    def load_policy(self, path, policies):
+        """
+        Loads a YAML file and prompts scheduling updates
+        :param path: Path to YAML file on disk
+        :param policies: Dictionary of policies to update
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+                new_policies = PolicyCollection.from_data(policy_config, self.options)
+
+                if new_policies:
+                    for p in new_policies:
+                        log.info("Loading Policy %s from %s" % (p.name, path))
+
+                        p.validate()
+                        policies.update({p.name: {'policy': p}})
+
+                        # Update periodic and set event update flag
+                        self.update_periodic(p)
+                        if p.data.get('mode', {}).get('events'):
+                            self.require_event_update = True
+
+            except Exception as exc:
+                log.error('Invalid policy file %s %s' % (path, exc))
```

Comment:
generally daemons refuse to start / exit when they have invalid config files.

Reply:
In this case it might have 300 good policies it is already executing, then 1 bad policy shows up, so we can't load the bad one but we'll keep on running.  After the user fixes that policy it will get loaded then.